pygfx/materials/_compat.py

```diff
@@ -144,3 +144,132 @@ def material_from_trimesh(x):
         raise NotImplementedError(f"Conversion of {type(x)} is not supported.")
 
     return gfx_material
+
+
+def texture_from_numpy_array(image_array, dim=2, **kwargs):
+    """Convert a numpy array representing an Open3D image to a pygfx Texture.
+
+    Parameters
+    ----------
+    image_array : numpy.ndarray
+        The numpy array representing the image data.
+        The expected shape is (height, width, channels).
+    dim : int
+        The number of spatial dimensions of the image.
+    kwargs : Any
+        Additional kwargs are forwarded to :class:`pygfx.Texture`.
+
+    Returns
+    -------
+    image_texture : Texture
+        A texture object representing the given image.
+
+    """
+
+    import numpy as np
+    from pygfx import Texture
+    import open3d.cpu.pybind.geometry
+
+    # Check if it is an open3d image
+    if isinstance(image_array, open3d.cpu.pybind.geometry.Image):
+        image_array = np.ascontiguousarray(image_array)
+
+    # Ensure input is a valid numpy array
+    if not isinstance(image_array, np.ndarray):
+        raise NotImplementedError("Input must be a numpy array")
+
+    # Ensure the input array has 2D (height, width) or 3D (height, width, channels) shape
+    if image_array.ndim not in [2, 3]:
+        raise ValueError("Input array must be a 2D or 3D numpy array")
+
+    # Normalize and adjust channels if needed
+    if image_array.ndim == 2:
+        # Assume single-channel grayscale image
+        buffer_shape = (*image_array.shape, 1)
+        image_array = image_array.reshape(buffer_shape)
+
+    # Create a memoryview of the data for the texture
+    m = memoryview(image_array)
+
+    return Texture(m, dim=dim, **kwargs)
+
+
+def material_from_open3d(x):
+    """Convert an Open3D MaterialRecord object into a pygfx material.
+
+    Parameters
+    ----------
+    x : open3d.visualization.rendering.MaterialRecord | open3d.geometry.Geometry3D
+        Either the actual Open3D MaterialRecord or an object containing
+        a material (e.g., geometry).
+
+    Returns
+    -------
+    converted : Material
+        The converted pygfx material.
+
+    """
+
+    import open3d as o3d
+    from pygfx.materials import MeshPhongMaterial, MeshStandardMaterial
+    from pygfx import Color
+    import numpy as np
```

Comment:
Makes sense to put the `open3d` import here, so its only loaded when needed, but the other imports can be at the top of the module.

Reply:
Yes, some of the imports are anyways already imported. I have refactored it.